doc/user-guide.md

```diff
@@ -1 +1,384 @@
+  user model, and are not required to meet the goal of creating a mono-repo.
+
+### Value-added behavior
+
+To support this model, git-meta commands offer the following conveniences not
+supported by normal Git commands:
+
+- the ability to operate on the entire mono-repo when the current working
+  directory is within a submodule, e.g.:
+
+```bash
+$ git meta open my-sub-repo
+$ cd my-sub-repo/foo/bar
+$ git checkout -b a-new-branch       # makes a branch in the sub-repo
+$ git meta checkout -b a-new-branch  # makes a branch in the meta-repo
+```
+
+- the ability to target paths within any open submodule regardless of current
+  working directory, e.g.:
+
+```bash
+$ echo >> sub-repo-x/README.md
+$ cd sub-repo-y
+$ git meta add ../sub-repo-x/README.md
+```
+
+- the ability to target multiple sub-repos in one command:
+
+```bash
+$ git meta commit sub-repo-x/README.md sub-repo-y/foo/bar/main.cpp
+```
+
+or even by directory tree:
+
+```bash
+$ echo "a change" >> my-product/sub-repo-a/README.md
+$ echo "another change" >> my-product/sub-repo-b/README.md
+$ git meta add my-product
+```
+
+- removes the need to explicitly stage and commit updated submodules:
+
+Normally in Git, if you add a commit to a submodule you need to stage the
+submodule too:
+
+```bash
+$ cd my-sub-repo
+$ echo >> foo
+$ git commit -m "changed foo" foo
+$ cd ..
+$ git commit -m "changed my-sub-repo"
+On branch master
+Changes not staged for commit:
+        modified:   my-sub-repo (new commits)
+
+no changes added to commit
+```
+
+This process is greatly simplified with git-meta:
+
+```bash
+$ echo >> my-sub-repo/foo
+$ git meta-commit -m "changed foo" my-sub-repo/foo
+```
+
+Even if you want to explicitly make two commits, you do not need to manually
+stage the sub-repo:
+
+```bash
+$ cd my-sub-repo
+$ echo >> foo
+$ git commit -m "changed foo" foo
+$ git meta commit -m "changed my-sub-repo"
+```
+
+## Usage Scenarios
+
+### Cloning
+
+We do not provide a git-meta command for cloning as the built-in Git command
+does exactly the right thing:
+
+```bash
+$ git clone http://example.com/your-meta-repo.git meta
+```
+
+### Submodule Visibility
+
+A freshly-cloned meta-repo is usually empty, containing a tree of empty
+sub-directories where submodules are mounted.  The first thing you'll want to
+do is to make sub-projects available:
+
+```bash
+$ git meta open my-team/my-project
+```
+
+You can open a whole tree of sub-repos:
+
+```bash
+$ git meta open your-team/
+```
+
+There reverse operation is `close`:
+
+```bash
+$ git meta close your-team
+```
+
+The primary advantage of `git meta open` over `git submodule update --init`
+(besides a cleaner syntax) is that it will propagate repository templates into
+opened submodules (see `git meta open --help` for more information).  The
+`close` command is provided for symmetry.
+
+
+### Switching Branches
+
+We provide `git meta checkout` to switch branches:
+
+```bash
+$ git meta checkout my-feature
+```
+
```

Comment:
i think providing the raw git version of each of these commands (in addition to the prose descriptions) might help git experts understand more quickly.